pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/minion/TaskMetricsEmitter.java

```diff
@@ -86,5 +96,15 @@ protected final void runTask() {
         LOGGER.error("Caught exception while getting metrics for task type {}", taskType, e);
       }
     }
+
+    // Emit metric to count the number of online minion instances.
+    List<String> onlineInstances = _pinotHelixResourceManager.getOnlineInstanceList();
+    int onlineMinionInstanceCount = 0;
+    for (String onlineInstance : onlineInstances) {
+      if (onlineInstance.startsWith(CommonConstants.Helix.PREFIX_OF_MINION_INSTANCE)) {
+        onlineMinionInstanceCount++;
+      }
+    }
+    _controllerMetrics.addValueToGlobalGauge(ControllerGauge.ONLINE_MINION_INSTANCES, onlineMinionInstanceCount);
```

Comment:
This task will run on the controller that hashes into the task name. We should be emitting this metric from one controller only. So, either we introduce another task, or change the task to be running on one controller (that is fine too, we don't expect that there are too many task types in the clustter)

Reply:
This metric is emitted only from only one controller, right? Suppose no controller is added or removed, there will be only one controller and it's always that controller which will run the actual logic of this task and emit this metric.

Reply:
Ah, u r right. My bad. I thought we hashed on task type. it turns out we hashed on task name, so this will be emitted only on one controller.